File: linked_list.cpp
#include<iostream>
#include<cstdlib>
#define n 10
using namespace std;

struct Node{
    Node* prev;
    int num;
    Node* next;
};

class list{
private:
    Node *head, *tail;
    
public:
    void createnode(int value);
    void insert(int val);
    void display();
    void sort();
    int length();
    list(){
        head = NULL;
        tail = NULL;
    };
};

void list::createnode(int value){
    Node* N = new Node;
    N->num = value;
    N->next = NULL;
    if(head == NULL)
    {
        head = N;
        tail = N;
    }
    else
    {
        tail->next = N;
        tail = N;
    }
}

void list::display(){
    Node* temp = new Node;
    temp=head;
    while(temp != NULL)
    {
        cout<<temp->num<<'\t';
        temp = temp->next;
    }
}

void list::sort(){
    Node* t1;
    if(head == NULL || head->next == NULL)
        return;
    else
        t1 = head;
    while(t1 != NULL)
    {
        Node* t2;
        t2 = head;
        while(t2->next != t1->next)
        {
            if(t2->num > t1->num)
            {
                int temp = t2->num;
                t2->num = t1->num;
                t1->num = temp;
            }
            t2 = t2->next;
        }
        t1 = t1->next;
        //delete t2;
    }
    
}

void list::insert(int val){
    Node* t = new Node;
    t->num = val;
    tail->next = t;
    tail = t;
}
int list::length(){
    int len = 0;
    if(head == NULL)
        return 0;
    Node* l = new Node;
    l = head;
    while(l != NULL)
    {
        len++;
        l = l->next;
    }
    return len;
}

int main()
{
    list mine;
    
    int a[] = {1,2,4,7,1,4,10,13,5};
    
    for(int j=0;j<9;j++)
        mine.createnode(a[j]);
    
    mine.insert(20);
    mine.display();
    //mine.sort();
    cout<<'\n';
    //mine.display();
    //cout<<"the length of the list is "<<mine.length();
    return 0;
}

File: linkedlist.java
package com.example.helloworld;

public class linkedlist {
    node head;
    static class node
    {
        int num;
        node next;
        node(int n) {this.num = n;}
    }

    public static void printer(linkedlist a)
    {
        node temp = a.head;
        if(temp == null)
        {
            System.out.print("NUll");
        }
        while (temp != null)
        {
            System.out.print(temp.num);
            System.out.print(" ");
            temp = temp.next;
        }
        System.out.print("\n");

        return;
    }

    public static linkedlist insert(linkedlist a, int n)
    {
        node nex = new node(n);
        nex.next = null;
        if(a.head == null)
        {
            a.head = nex;
        }
        else
        {
            node temp = a.head;
            while(temp.next != null)
            {
                temp = temp.next;
            }
            temp.next = nex;
        }

        return a;
    }

    public static void swap(node a, node b)
    {
        int temp = a.num;
        a.num = b.num;
        b.num = temp;
    }

    public static linkedlist sorter(linkedlist a)
    {
        node temp = a.head;
        if(temp.next == null)
        {
            return a;
        }
        while(temp.next != null)
        {
            node temp2 = temp.next;
            while(temp2 != null)
            {
                if(temp.num > temp2.num)
                {
                    swap(temp,temp2);
                }
                temp2 = temp2.next;
            }

            temp = temp.next;
        }
        return a;
    }

    public static void main(String[] args) {
        linkedlist lister = new linkedlist();
        lister = insert(lister,10);
        lister = insert(lister,1);
        lister = insert(lister,100);
        lister = insert(lister,2);
        lister = insert(lister,50);
        lister = insert(lister,60);
        lister = insert(lister,40);
        lister = insert(lister,35);
        lister = insert(lister,3);
        lister = insert(lister,400);
        lister = insert(lister,200);
        lister = insert(lister,9);
        lister = insert(lister,15);
        lister = insert(lister,16);
        lister = insert(lister,9);

        printer(lister);
        node a = lister.head;
        swap(a,a.next);
        printer(lister);
        lister = sorter(lister);
        printer(lister);
    }
}
